### PR TITLE
params for list end point.

### PR DESCRIPTION
**Pull request by fabricematrat**

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/137/
Test PASSed.

File: csclient/params/params.go [hunk 1]
```diff
@@ -173,6 +173,23 @@ type SearchResponse struct {
...
+	// specified in the include flags, holding the
+	// data that would be returned by reading /meta/meta?id=id.
+	// Metadata not relevant to a particular result will not
+	// be included.
+	Meta map[string]interface{} `json:",omitempty"`
+}
+
+// ListResponse holds the response from a list operation.
+type ListResponse struct {
+	Total      int
```
Comment (mhilton) [inline csclient/params/params.go]: In search this shows the total number of matches even if they aren't all listed, what does it do here?

**Comment by mhilton** [comment]
:+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/138/
Test PASSed.

File: csclient/params/params.go [hunk 2]
```diff
@@ -173,6 +173,22 @@ type SearchResponse struct {
...
+	// specified in the include flags, holding the
+	// data that would be returned by reading /meta/meta?id=id.
+	// Metadata not relevant to a particular result will not
+	// be included.
+	Meta map[string]interface{} `json:",omitempty"`
+}
+
+// ListResponse holds the response from a list operation.
+type ListResponse struct {
+	Results    []ListResult
```
Comment (rogpeppe) [inline csclient/params/params.go]: How about just:

    Results []SearchResult

?
I don't think there's any particular need for SearchResult and ListResult to be different types.

**Comment by rogpeppe** [comment]
LGTM with one suggestion.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/139/
Test FAILed.

File: csclient/params/params.go [hunk 3]
```diff
@@ -155,8 +155,8 @@ type Published struct {
 // debugstatus.CheckResult directly.
 type DebugStatus debugstatus.CheckResult
 
-// SearchResult holds a single result from a search operation.
-type SearchResult struct {
+// EntityResult holds a single result from entities lookup.
```
Comment (mhilton) [inline csclient/params/params.go]: It will contain the resolved entity ID along with any requested metadata.  ?

**Comment by mhilton** [comment]
+1 with a bit more comment (take or leave)

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/140/
Test FAILed.

File: csclient/params/params.go [hunk 4]
```diff
@@ -155,8 +155,8 @@ type Published struct {
 // debugstatus.CheckResult directly.
 type DebugStatus debugstatus.CheckResult
 
-// SearchResult holds a single result from a search operation.
-type SearchResult struct {
+// EntityResult holds a the resolved entity ID along with any requested metadata.
```
Comment (rogpeppe) [inline csclient/params/params.go]: s/the //

**Comment by rogpeppe** [comment]
LGTM. Even though the change is backwardly incompatible you can actually make the test build pass (and avoid pressing the big green button) by changing the deps so that this depends on your not-yet-committed version of the charmstore (which you'd need to push up to a temporary branch in the charmstore repo).

Do that only when you're ready to land the charmstore changes, then land charmstore depending on the landed version of charmrepo, then update charmrepo to use the just-landed version of charmstore.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/141/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/142/
Test PASSed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmrepo/143/
Test PASSed.

**Comment by fabricematrat** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmrepo-merge

**Pull request merged**